File: frontend/lib/features/session/domain/recent_intervention.dart
import 'task_suggestion.dart';

/// A previously suggested task, used so the AI can vary its category
/// choice across the last several interventions, not just the single
/// most recent one.
class RecentIntervention {
  const RecentIntervention({required this.title, required this.category});

  final String title;
  final TaskCategory category;
}

File: backend/app/domain/recent_intervention.py
from dataclasses import dataclass

from app.domain.task_suggestion import TaskCategory


@dataclass(frozen=True)
class RecentIntervention:
    """A previously suggested task, sent inline by the caller so the AI
    can vary its category choice across the last several interventions,
    not just the single most recent one."""

    title: str
    category: TaskCategory
